src/Glpi/Asset/CustomField.php

```diff
@@ -0,0 +1,363 @@
+                _n('Asset', "Assets", Session::getPluralNumber()) => array_combine(
+                    $CFG_GLPI['asset_types'],
+                    array_map(static fn ($t) => $t::getTypeName(1), $CFG_GLPI['asset_types'])
+                ),
+            ];
+            $allowed_dropdown_itemtypes = array_merge_recursive($allowed_dropdown_itemtypes, Dropdown::getStandardDropdownItemTypes());
+        }
+
+        return $allowed_dropdown_itemtypes;
+    }
+
+    /**
+     * Get the list of field types available
+     * @return array<string, class-string<TypeInterface>>
+     */
+    public static function getFieldTypes(): array
+    {
+        // Get files in the current directory
+        $directory_iterator = new DirectoryIterator(__DIR__ . '/CustomField');
+        static $all_types = null;
+
+        if ($all_types === null) {
+            $all_types = [];
+            /** @var \SplFileObject $file */
+            foreach ($directory_iterator as $file) {
+                // Compute class name with the expected namespace
+                $classname = $file->getExtension() === 'php'
+                    ? 'Glpi\\Asset\\CustomField\\' . $file->getBasename('.php')
+                    : null;
+
+                // Validate that the class is a valid question type
+                if (
+                    $classname !== null
+                    && class_exists($classname)
+                    && is_subclass_of($classname, TypeInterface::class)
+                    && (new ReflectionClass($classname))->isAbstract() === false
+                ) {
+                    $all_types[$classname::getKey()] = $classname;
+                }
+            }
+        }
+        return $all_types;
+    }
+
+    public function getTabNameForItem(CommonGLPI $item, $withtemplate = 0)
+    {
+        $count = 0;
+        if ($_SESSION['glpishow_count_on_tabs']) {
+            $count = countElementsInTable(self::getTable(), [
+                AssetDefinition::getForeignKeyField() => $item->fields['id'],
+            ]);
+        }
+        return self::createTabEntry(self::getTypeName(), $count);
+    }
+
+    public static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $withtemplate = 0)
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        if (!$item->canViewItem()) {
+            return false;
+        }
+
+        $canedit = $item->canUpdateItem();
+        $rand = mt_rand();
+        if ($canedit) {
+            TemplateRenderer::getInstance()->display('components/form/viewsubitem.html.twig', [
+                'cancreate' => self::canCreate(),
+                'id'        => $item->fields['id'],
+                'rand'      => $rand,
+                'type'      => self::class,
+                'parenttype' => self::$itemtype,
+                'items_id'  => self::$items_id,
+                'add_new_label' => __('Add a new field'),
+                'datatable_id' => 'datatable_customfields' . $rand,
+            ]);
```

Comment:
I do not really like the "inlined" form handling, but it can be reviewed in another PR. The fact is that in GLPI, forms are usually displayed in dedicated pages, and doing it differently will probably result in having to use mode dedicated code (and often it is JS code that only a few of us are comfortable with). I am also not sure that it provides a better UX.

Reply:
There are other cases where this is done and I seem to remember inlining some forms during sublist Twig PRs. In this case, custom field definitions have no dedicated form page. They are added/edited/deleted only from here.